### ISSUE TITLE
Proxy is not working with CryptoExchange.Net 6.2.1 and Binance.net 9.1.6

### ISSUE THREAD

**Issue by jdponomarev** [JKorf/Binance.Net#1317]
**Describe the bug**
Since I upgraded to CryptoExchange.Net 6.2.1 and Binance.net 9.1.6, my traffic is not coming through proxy anymore.
Same issue for Kucoin.net and Kraken.net after I updated to the latest version, no traffic going through my proxy.

**To Reproduce**
Initialise Binance like this, with proxy credentials that are 100% wrong:
`
BinanceRestClient.SetDefaultOptions(options =>
{
        options.Proxy = new ApiProxy("123", 123, "123", "123");
});
services.AddScoped<IBinanceRestClient, BinanceRestClient>();
`

**Expected behavior**
This should fail every for call, but it still works, even thought I can see Client Options have Proxy on the client. If I set up correct proxy details it is expected to use the proxy, but it doesn't

**Comment by jdponomarev** [comment]
hey @JKorf I've tried to create a minimal example project with this issue, but couldn't...
A minimal example like yours works for me.

In my project it just does pick it up when configured in ConfigureServices at Startup.cs with BinanceRestClient.SetDefaultOptions. I can see other options are set up correctly, and the proxy is actually set too, but is not used and doesn't throw errors on invalid configuration.

I actually ended up creating clients like this and it works perfect for me:

```
_binanceClient = new BinanceRestClient((options =>
{
    options.Environment = BinanceEnvironment.Live;
    options.ApiCredentials = new ApiCredentials(binanceSettings.ApiKey, binanceSettings.ApiSecret);
    options.Proxy = new ApiProxy(configuration["Proxy:host"], int.Parse(configuration["Proxy:port"]),
            configuration["Proxy:login"], configuration["Proxy:password"]);
}));
```

I'm using other CryptoExchange.Net libraries too (all at latest versions). I'll let you know if I'm able to figure it out
